pkg/scheduler/state/state.go

```diff
@@ -271,7 +272,7 @@ func (s *stateBuilder) State(reserved map[types.NamespacedName]map[string]int32)
 				return err == nil, nil
 			})
 
-			if pod != nil {
+			if pod != nil && schedulablePods.Has(OrdinalFromPodName(pod.GetName())) {
```

Comment:
Additional check is for cases where a [newly] unschedulable pod is part of the existing placements list? Can ever be a pending pod because of the ordinality. Must be a case where a pod is going to be evicted?

Reply:
It's both case an existing evicted and then unschedulable pod or an newly unschedulable pod.